tests/test_qsimh_device.py

```diff
@@ -0,0 +1,400 @@
+# Copyright 2018 Xanadu Quantum Technologies Inc.
```

Comment:
Same with these tests. They have been copied from `test_cirq_device.py`, with only some minor modifications to work with `QSimhDevice`. In addition, two tests for the `qsimh_options` kwarg has been added.